pkg/store/bucket.go

```diff
@@ -769,6 +774,10 @@ func blockSeries(
 		return nil, nil, errors.Wrap(err, "preload chunks")
 	}
 
+	if err := chunksSizeLimiter.Reserve(uint64(chunkr.stats.seriesFetchedSizeSum)); err != nil {
```

Comment:
As per your question:

> Another question is is it a proper way to limit based on a chunk reader stats? https://github.com/thanos-io/thanos/pull/3089/files#diff-a75f50a9f5bf5b21a862e4e7c6bd1576R777
Or is it better to pass the limiter to the preload function?

That's a good one, it really depends on one thing: Do we want to return partial data if limit is exceeded or fail everything? (:

Reply:
My take on that is that a user might be surprised when find out that the returned response was partial. 
It might make sense to create a config for that and apply for both: limiting by a number of samples and by byte size.

Reply:
Oh yes, we have warnings for that. Everything with warning is assumed partial - then querier decide if it's error or not with partialResponseStrategy request option (: